tools/ci_build/build.py

```diff
@@ -1627,6 +1628,14 @@ def generate_build_tree(
     if args.use_snpe:
         cmake_args += ["-Donnxruntime_USE_SNPE=ON"]
 
+    targeting_arm64 = args.arm64
+    if not targeting_arm64:
+        if is_macOS():
+            targeting_arm64 = args.osx_arch == "arm64"
+        else:
+            targeting_arm64 = platform.machine() in ["aarch64", "arm64", "ARM64"]
```

Comment:
will `targeting_arm64` be properly set for Android?

I wonder if it is simpler to just directly set `onnxruntime_USE_KLEIDIAI` to `"ON" if not args.no_kleidiai else "OFF"`.